llvm/tools/ir2builder/ir2builder.cpp

```diff
@@ -0,0 +1,1676 @@
+        else
+          return "/* Unknown type in data sequential constant */";
+        First = false;
+      }
+      if (CDS->getNumElements() > 1)
+        Values += "}";
+      Values += ")";
+    }
+
+    return formatv("{0}{1}::get({2}, {3})", LLVMPrefix, ClassName, Ctx, Values);
+  }
+  if (isa<ConstantAggregateZero>(C))
+    return formatv("{0}ConstantAggregateZero::get({1})", LLVMPrefix,
+                   asStr(C->getType()));
+  if (isa<PoisonValue>(C))
+    return formatv("{0}PoisonValue::get({1})", LLVMPrefix, asStr(C->getType()));
+  if (isa<UndefValue>(C))
+    return formatv("{0}UndefValue::get({1})", LLVMPrefix, asStr(C->getType()));
+  if (auto *BA = dyn_cast<BlockAddress>(C))
+    return formatv("{0}BlockAddress::get({1}, {2})", LLVMPrefix,
+                   asStr(BA->getFunction()), asStr(BA->getBasicBlock()));
+  if (isa<ConstantPointerNull>(C))
+    return formatv("{0}ConstantPointerNull::get({1})", LLVMPrefix,
+                   asStr(C->getType()));
+  if (auto *CTN = dyn_cast<ConstantTargetNone>(C)) {
+    auto CTNType = CTN->getType();
+
+    std::string TypeStr = "{";
+    bool First = true;
+    for (unsigned I = 0; I < CTNType->getNumTypeParameters(); ++I) {
+      if (!First)
+        TypeStr += ", ";
+      TypeStr += asStr(CTNType->getTypeParameter(I));
+      First = false;
+    }
+    TypeStr += "}";
+
+    std::string IntsStr = "{";
+    First = true;
+    for (unsigned I = 0; I < CTNType->getNumIntParameters(); ++I) {
+      if (!First)
+        IntsStr += ", ";
+      IntsStr += std::to_string(CTNType->getIntParameter(I));
+      First = false;
+    }
+    IntsStr += "}";
+
+    return formatv("{0}ConstantTargetNone::get({0}TargetExtType::get({1}, "
+                   "\"{2}\", {3}, {4}))",
+                   LLVMPrefix, Ctx, escape(CTNType->getName().str()), TypeStr,
+                   IntsStr);
+  }
+  if (isa<ConstantTokenNone>(C)) {
+    return formatv("{0}ConstantTokenNone::get({1})", LLVMPrefix, Ctx);
+  }
+  if (auto *CE = dyn_cast<ConstantExpr>(C)) {
+    (void)CE;
+    return "/* TODO: ConstantExpr creation */";
+    // TODO: Dunno how to create this... Fails either on out of range or
+    // even on incorrect opcode.
+    // return LLVMPrefix + "ConstantExpr::get(" +
+    // std::to_string(CE->getOpcode()) + ", " +
+    //       asStr(CE->getOperand(0)) + ", " + asStr(CE->getOperand(1)) + ")";
+  }
+  if (auto *CE = dyn_cast<ConstantPtrAuth>(C)) {
+    (void)CE;
+    return "/* TODO: ConstantPtrAuth value creation */";
+  }
+  if (auto *CE = dyn_cast<DSOLocalEquivalent>(C)) {
+    (void)CE;
+    return "/* TODO: DSOLocalEquivalent value creation */";
+  }
+  if (auto *CE = dyn_cast<NoCFIValue>(C)) {
+    (void)CE;
+    return "/* TODO: NoCFIValue value creation */";
+  }
+  if (auto *CE = dyn_cast<GlobalValue>(C)) {
+    // This should not really happen as asStr for Value should be always called.
+    return asStr(cast<Value>(CE));
```

Comment:
Since you're casting to a base class you can use `static_cast`. `cast` contains an assert to check the cast is valid, that's for casting to a derived class.